pkg/storage/etcd3/compact.go

```diff
@@ -63,11 +67,26 @@ func StartCompactor(ctx context.Context, client *clientv3.Client) {
 }
 
 // compactor periodically compacts historical versions of keys in etcd.
-// After compaction, old versions of keys set before given interval will be gone.
-// Any API call for the old versions of keys will return error.
+// It will compact old version of keys set before given interval time.
+// In other words, after compaction, it will only contain keys set during last interval.
+// Any API call for the older versions of keys will return error.
 // interval: the interval between each compaction. The first compaction happens after "interval".
 func compactor(ctx context.Context, client *clientv3.Client, interval time.Duration) {
-	var curRev int64
+	// compactRevKey's value will be set to the string of compacted revision.
+	// compactRevKey's version will be used as logical time for comparison.
+	// We refer to compactRevKey's version as compact time.
+	// Initially, because the key doesn't exist, compact time is 0.
+	//
+	// Algorithm:
+	// - Compare to see if local compact_time = remote compact_time.
+	// - If yes, increment both local and remote compact_time, and do a compaction.
+	// - If not, set local to remote compact_time.
+	//
+	// During failover, we will have guarantee that another apiserver will compact after > given compact_interval,
```

Comment:
I'm slightly confused by this failover statement, b/c apiservers are active-active.